src/decoders/mod.rs

```diff
@@ -186,12 +186,66 @@ impl Camera {
   }
 }
 
-pub fn ok_image(camera: &Camera, width: usize, height: usize, wb_coeffs: [f32;4], image: Vec<u16>) -> Result<RawImage,String> {
-  Ok(RawImage::new(camera, width, height, wb_coeffs, image))
+/// Possible values for the IFD tag Orientation (0x0112)
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum Orientation {
+  Normal,
+  HorizontalFlip,
+  Rotate180,
+  VerticalFlip,
+  Transpose,
+  Rotate90,
+  Transverse,
+  Rotate270,
+  Unknown,
 }
 
-pub fn ok_image_with_blacklevels(camera: &Camera, width: usize, height: usize, wb_coeffs: [f32;4], blacks: [u16;4], image: Vec<u16>) -> Result<RawImage,String> {
-  let mut img = RawImage::new(camera, width, height, wb_coeffs, image);
+impl Orientation {
+  /// Convert a u16 into its corresponding enum value
+  pub fn from_u16(orientation: u16) -> Orientation {
+    match orientation {
+      1 => Orientation::Normal,
+      2 => Orientation::HorizontalFlip,
+      3 => Orientation::Rotate180,
+      4 => Orientation::VerticalFlip,
+      5 => Orientation::Transpose,
+      6 => Orientation::Rotate90,
+      7 => Orientation::Transverse,
+      8 => Orientation::Rotate270,
+      _ => Orientation::Unknown,
+    }
+  }
+
+  /// Extract orientation from given EXIF data
+  pub fn from_tiff_ifd(tiff: &TiffIFD) -> Orientation {
+    // Lookup the orientation of the image to store for rotating after demosaic
+    if let Some(ifd) = tiff.find_first_ifd(Tag::Orientation) {
+      // Since we got here it should always be safe to unwrap
+      Orientation::from_u16(ifd.find_entry(Tag::Orientation).unwrap().get_usize(0) as u16)
+    } else {
+      Orientation::Unknown
+    }
+  }
+
+  /// Return true if orientation results in a transpose, i.e. width and height
+  /// are swapped
+  pub fn is_transpose(&self) -> bool {
```

Comment:
My knowledge of rust enums is still quite incomplete. Can't you have the same named enums but have them be instances of the same generic 3 bool item? If not the conversion one seems fine. I'd call it something like OrientationFlips instead of ImageTransformation.